cmake/dependencies.cmake

```diff
@@ -0,0 +1,109 @@
+# Download all external projects in the specified directory.
+include(ExternalProject)
+set(EP_PREFIX "${PROJECT_SOURCE_DIR}/external")
+set_directory_properties(PROPERTIES
+  EP_PREFIX ${EP_PREFIX}
+)
+
+include(cmake/DownloadProject.cmake)
+
+# Project dependencies.
+find_package(Threads REQUIRED)
```

Comment:
Add `set(CMAKE_THREAD_PREFER_PTHREAD ON)` before this line, just in case if other threading libraries are available in the system.